### ISSUE TITLE
QnA maker doesn't run 

### ISSUE THREAD

**Issue by CathyQian** [Microsoft/BotBuilder-Samples#566]
## Sample information
1. Sample type: Sample
2. Sample language: C#
3. Sample name: 11.QnAMaker

## Describe the bug
Following the procedure to run the bot in both Visual Studio and Visual Studio Code resulted in errors.

In Visual Studio, the bot can be built and local host can be created. However, when I enter text into the Bot Emulator, it says "couldn't send".

In Visual Studio Code, I went to the right folder in terminal and typed "dotnet run". The bot can't be built and I received the following error messages:
"""
Using launch settings from D:\Cloned_Repo\BotBuilder-Samples\samples\csharp_dotnetcore\11.qnamaker\Properties\launchSettings.json...
warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to bind to http://localhost:3978 on the IPv4 loopback interface: 'Error -4092 EACCES permission denied'.
warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to bind to http://localhost:3978 on the IPv6 loopback interface: 'Error -4092 EACCES permission denied'.
crit: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to start Kestrel.
System.IO.IOException: Failed to bind to address http://localhost:3978. ---> System.AggregateException: One or more errors occurred. (Error -4092 EACCES permission denied) (Error -4092 EACCES permission denied) ---> Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.UvException: Error -4092 EACCES permission denied
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.LibuvFunctions.ThrowError(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.LibuvFunctions.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Networking.UvTcpHandle.Bind(IPEndPoint endPoint)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Listener.ListenTcp(Boolean useFileHandle)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Listener.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Listener.<>c.<StartAsync>b__8_0(Listener listener)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.LibuvThread.CallbackAdapter`1.<>c.<.cctor>b__3_1(Object callback, Object state)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.LibuvThread.DoPostWork()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.ListenerPrimary.<StartAsync>d__15.MoveNext()
"""

Anyone has run the QnAMaker sample successfully yet? Thanks!

**Comment by robhellest** [comment]
Hi Cathy - close to running successfully; if trying to run QnA Maker in Bot Framework Emulator, with Visual Studio solution running locally in IIS - following error is received: **System.ArgumentException: 'Invalid configuration. Please check your '.bot' file for a QnA service named 'QnABot'.'**
Best regards - Rob

**Comment by CathyQian** [comment]
@robhellest Yes, that's exactly the error I'm receiving. So what is the solution? Thanks!

**Comment by CathyQian** [comment]
@robhellest @JonathanFingold and I finally made it work. Here is what we did:
Change the BotConfiguration.bot file content to the following:

{
  "name": "QnABotSample",
  "services": [
    {
      "type": "endpoint",
      "name": "development",
      "endpoint": "http://localhost:3978/api/messages",
      "appId": "",
      "id": "1",
      "appPassword": ""
    },
    {
      "type": "qna",
      "name": "QnABot",
      "KbId": "xxxx",
      "Hostname": "xxxxxx",
      "EndpointKey": "xxx"
    }
  ],
  "version": "2.0",
  "padlock": ""
}


Fill in the KbId, Hostname and EndpointKey of  your QnA maker. Make sure the second name in the services section (I use "QnABot") is the name of your QnA maker. Also, go to your QnABot.cs file, make sure the QnAMakerKey string is also the name of your QnA maker. In my case, I have the following line:
public static readonly string QnAMakerKey = "QnABot";
